docs/db.md

```diff
@@ -1,4 +1,200 @@
+In this section of the docs, to have a clearer understanding of how one would use `mu-haskell` to talk to a database, we are going to have a walk through the example of [`with-persistent`](https://github.com/higherkindness/mu-haskell/tree/master/examples/with-persistent).
+
+## First steps
+
+We are going to start with our source of truth: the proto file.
+
+```protobuf
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package withpersistent;
+
+message PersonRequest { int64 identifier = 1; }
+message Person { PersonRequest pid = 1; string name = 2; int32 age = 3; }
+
+service PersistentService {
+  rpc getPerson (PersonRequest) returns (Person);
+  rpc newPerson (Person) returns (PersonRequest);
+  rpc allPeople (google.protobuf.Empty) returns (stream Person);
+}
+```
+
+Maybe this example looks a bit contrived but bear with me, it covers a common use case when working with protobuf: that one of the messages has another message as its identifying key. Easier examples would not suffer from this and should be easier to implement.
+
+## Definning our Schema
+
+You are going to need to enable the following extensions:
+
+```haskell
+{-# language DataKinds                  #-}
+{-# language DeriveGeneric              #-}
+{-# language DerivingVia                #-}
+{-# language DuplicateRecordFields      #-}
+{-# language FlexibleContexts           #-}
+{-# language FlexibleInstances          #-}
+{-# language GADTs                      #-}
+{-# language GeneralizedNewtypeDeriving #-}
+{-# language MultiParamTypeClasses      #-}
+{-# language OverloadedStrings          #-}
+{-# language PolyKinds                  #-}
+{-# language QuasiQuotes                #-}
+{-# language ScopedTypeVariables        #-}
+{-# language StandaloneDeriving         #-}
+{-# language TemplateHaskell            #-}
+{-# language TypeFamilies               #-}
+{-# language TypeOperators              #-}
+{-# language UndecidableInstances       #-}
+```
+
+As we've seen in the rest of the docs, we define our own data types to mirror our protobuf schema:
+
+```haskell
+grpc "PersistentSchema" id "with-persistent.proto"
+
+newtype MPersonRequest = MPersonRequest
+  { identifier :: Maybe Int64
+  } deriving (Eq, Show, Ord, Generic)
+
+instance ToSchema   Maybe PersistentSchema "PersonRequest" MPersonRequest
+instance FromSchema Maybe PersistentSchema "PersonRequest" MPersonRequest
+
+data MPerson = MPerson
+  { pid  :: Maybe MPersonRequest
+  , name :: Maybe T.Text
+  , age  :: Maybe Int32 }
+  deriving (Eq, Ord, Show, Generic)
+
+instance ToSchema   Maybe PersistentSchema "Person" MPerson
+instance FromSchema Maybe PersistentSchema "Person" MPerson
+```
+
+Remember that all the magic starts with that first `grpc` line! ✨
+
+You might have noticed that this time, we are not using `DeriveAnyClass`, so we need to write the instances for `ToSchema` and `FromSchema` on a separate line from our deriving clause, and let GHC fill them for us.
```

Comment:
Maybe say that this choice is due to the [following Persistent bug](https://github.com/yesodweb/persistent/issues/578)?